### PR TITLE
Parse webhooks

### LINKED ISSUE #5: Integrate each added repo with github webhooks
When individuals create new boards, we'll want to setup each repo with github webhooks so we can make sure all the issues stay up to date

### PR DESCRIPTION
**Pull request by awaxman11**
Closes #5

### CODE REVIEW

File: lib/github_utils.rb [hunk 1]
```diff
@@ -62,5 +65,49 @@ def parse_issue(issue)
         :milestone_title   => issue[:milestone][:title]
       } : {})
     end
+
+    def update_issue(issue)
+      i = Issue.find_or_create_by(github_id: issue[:id])
+      parse_issue(issue)
```
Comment (CarlosPlusPlus) [inline lib/github_utils.rb]: L71 isn't actually doing anything, as that particular method returns a `{ }`, so will be removing this.

File: lib/github_utils.rb [hunk 2]
```diff
@@ -62,5 +65,49 @@ def parse_issue(issue)
...
+      clear_labels(i)
+      clear_custom_attributes(i)
+      add_labels(issue[:labels], i) unless issue[:labels].length == 0
```
Comment (CarlosPlusPlus) [inline lib/github_utils.rb]: I get the intention of L72-74, but I don't think this is the right approach. Clearing all labels and custom attributes every single time, then re-adding all labels and custom attributes is not the right approach. Need to sit down and really think about how we want to do this without crushing our DB with queries.

File: lib/github_utils.rb [hunk 3]
```diff
@@ -62,5 +65,49 @@ def parse_issue(issue)
...
+                        :repo_id => issue.repo_id,
+                        :name    => label[:name],
+                        :url     => label[:url],
+                        :color   => label[:color]
+                       )
+      end
+    end
+
+    def add_custom_attribute(label, issue)
+      label_type = label[:name].split(':').first
```
Comment (CarlosPlusPlus) [inline lib/github_utils.rb]: Here, we can use multi-parameter assignment to save us some cycles:

```ruby
type, value = label[:name].split(':')
```

File: lib/github_utils.rb [hunk 4]
```diff
@@ -62,5 +65,49 @@ def parse_issue(issue)
...
+    def clear_labels(issue)
+      issue.labels.clear
+    end
+
+    def clear_custom_attributes(issue)
+      ISSUE_CATEGORIES.each do |custom_category|
+        custom_category = 'issue_type' if custom_category == 'type'
+        custom_categories = {}
+        custom_categories[custom_category.to_sym] = nil
+        issue.update_attributes(custom_categories)
```
Comment (CarlosPlusPlus) [inline lib/github_utils.rb]: Remember, we need to minimize calls to the DB whenever possible.
Each call to `update_attributes` fires off a SQL query - not good man

Will refactor once I give this a bit more thought.

File: lib/github_utils.rb [hunk 5]
```diff
@@ -62,5 +65,49 @@ def parse_issue(issue)
...
+                        :url     => label[:url],
+                        :color   => label[:color]
+                       )
+      end
+    end
+
+    def add_custom_attribute(label, issue)
+      label_type = label[:name].split(':').first
+      if ISSUE_CATEGORIES.include?(label_type)
+        label_type = 'issue_type' if label_type == 'type'
```
Comment (CarlosPlusPlus) [inline lib/github_utils.rb]: @awaxman11 --- why did you call this `issue_type` in the DB?
Looks like it's OK to use the word `type` in the `Issue` model.

Comment (CarlosPlusPlus) [inline lib/github_utils.rb]: Answered my own question, looks like it's due to single table inheritance.

Will be disabling this for the `Issue` model per the following README:
http://stackoverflow.com/questions/7134559/rails-use-type-column-without-sti

**Comment by CarlosPlusPlus** [comment]
Will continue to work on the TODO's listed in here on this branch throughout the wk / wknd. Once I'm done here, I want to shift focus to tests before continuing with any more big features.

**Comment by awaxman11** [comment]
Just looked through the commits and they looks awesome on the surface. Haven't had a chance to test them out. Going to a concert tonight but will try and look this weekend. I also more than trust you if you just wanna merge it in :)

**Comment by CarlosPlusPlus** [comment]
Tested PR with @awaxman11, LGTM! :+1:

**Pull request merged**